### ISSUE TITLE
[Discussion] Azure Functions Runtime 2.0.12265

### ISSUE THREAD

**Issue by alrod** [Azure/app-service-announcements-discussions#79]
Discussion issue for the Azure Functions Runtime release 2.0.12265

**Comment by shibayan** [comment]
Please tell me the ETA of the deployment completion

**Comment by alrod** [comment]
@shibayan, it should be finished by the end of week (1/20).

**Comment by shibayan** [comment]
@alrod Thank you!

**Comment by Macros185** [comment]
This runtime version has caused all our functions to start throwing a new error when using Entity Framework.

> System.Data.SqlTypes.SqlNullValueException: Data is Null. This method or property cannot be called on Null values.

We had to pin our runtime version back to 2.0.12246 to resolve the issue. We assume updating all our NuGet packages to 2.2 will resolve this issue, but we haven't had the chance to check yet. Will update back here shortly.

Current EF Core package version is 2.1.4

**Comment by alrod** [comment]
@Macros185, can you please provide repro steps for the EF issue?

**Comment by tommyk** [comment]
@alrod the azure runtime version on windows recently bumped up (even higher than what this post is about), but the CLI tools did not get that update.  if you follow directions in official documentation for creating a new azure functions 2.0 project you are running locally as a dotnet 2.1 runtime, while in azure you are running on dotnet 2.2.  this is insanely frustrating and time consuming process to debug down all this just to get a simple `HttpClient` injected into my function.  running the azure tools cli off master branch got me to at lest have the same behavior (while its still an older version of the runtime).  

i have seen a few other posts across all these different repos complaining about what i think is this problem.  i hope you guys can get that CLI tools update to help a bunch of people out before they just give up and use something else.

**Comment by Macros185** [comment]
@alrod Sorry about the late reply. We found the cause for the SqlNullValueException we started experiencing with this update. This runtime started using a new EF Core version that caused some previously ignored bad code of ours to start throwing errors. A while back the "[Required]" attribute was manually added to a couple of properties in some of our EF entities (not sure why...). This used to be ignored, but a recent update to EF Core started using this attribute to force the value as required in the database which is not the case. We finally found and removed this attribute from our entities and the exceptions stopped. Just wanted to add this here in case anyone else experiences the same issue.
